### PR TITLE
[pentest] Configure alert handler for FI

### PR DESCRIPTION
**Pull request by nasahlpa**
This commit configures the alert handler to raise an interrupt that is handled by a custom ISR. This ISR registers the alert and write it back to the ot-sca host for further diagnosis.

The host code is located in https://github.com/lowRISC/ot-sca/pull/339

### CODE REVIEW

**Review by johannheyszl — APPROVED** [review]
Thanks Pascal this is very important for FI testing

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.c [hunk 1]
```diff
@@ -3,8 +3,141 @@
...
+/**
+ * Returns the registered alerts.
+ *
+ * Checks which of the picked alerts are registered by the alert handler. If
+ * the alert was registered, set a bit mask and clear the alert register.
+ */
+uint32_t sca_get_triggered_alerts(void) {
+  uint32_t reg_alerts = 0;
+
+  for (int32_t it = 0; it < NUM_ALERTS; it++) {
```
Comment (timothytrippel) [inline sw/device/tests/crypto/cryptotest/firmware/sca_lib.c:67]:
```suggestion
  for (size_t it = 0; it < NUM_ALERTS; it++) {
```

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.c [hunk 2]
```diff
@@ -3,8 +3,141 @@
...
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR);
+  CHECK_DIF_OK(dif_alert_handler_init(base_addr, &alert_handler));
+
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+
+  // Map all alerts to AlertHandlerClassA.
+  dif_alert_handler_class_t alert_classes[NUM_ALERTS];
+  for (int32_t it = 0; it < NUM_ALERTS; it++) {
```
Comment (timothytrippel) [inline sw/device/tests/crypto/cryptotest/firmware/sca_lib.c:106]:
```suggestion
  for (size_t it = 0; it < NUM_ALERTS; it++) {
```

**Review by timothytrippel — APPROVED** [review]

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.c (hunk 1 above)
Comment (nasahlpa) [reply sw/device/tests/crypto/cryptotest/firmware/sca_lib.c]: done, thx

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.c (hunk 2 above)
Comment (nasahlpa) [reply sw/device/tests/crypto/cryptotest/firmware/sca_lib.c]: done, thx

**Pull request merged**